### PR TITLE
Add function to install interceptors from list

### PR DESCRIPTION
**Pull request by dray92**
Signed-off-by: Debosmit Ray <debo@uber.com>

### CODE REVIEW

File: opentracing_instrumentation/client_hooks/__init__.py [hunk 1]
```diff
@@ -70,6 +70,25 @@ def install_patches(patchers='all'):
...
 
 
+def install_client_interceptors(client_interceptors=None):
+    """
+    Install client interceptors for the patchers.
+
+    :param client_interceptors: a list of client interceptors to install.
+        Should be a list of classes
+    """
+    if not type(client_interceptors) is list:
```
Comment (thomdixon) [inline opentracing_instrumentation/client_hooks/__init__.py]: I think you want something more like:
```lang=python
from collections.abc import Collection

if not isinstance(client_interceptors, Collection):
  ...
```

File: tests/opentracing_instrumentation/test_install_client_hooks.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from opentracing_instrumentation.interceptors import OpenTracingInterceptor
+
+
+class TestClientInterceptor(OpenTracingInterceptor):
+
+    def process(self, request, span):
+        pass
+
+
+def Any(cls):
```
Comment (thomdixon) [inline tests/opentracing_instrumentation/test_install_client_hooks.py]: Do you use this anywhere?

File: opentracing_instrumentation/client_hooks/__init__.py [hunk 3]
```diff
@@ -60,7 +64,7 @@ def install_patches(patchers='all'):
     if patchers is None or patchers == 'all':
         install_all_patches()
         return
-    if type(patchers) is list:
+    if _valid_args(patchers):
```
Comment (black-adder) [inline opentracing_instrumentation/client_hooks/__init__.py]: while you're here, do you mind reformatting this function to make it look like what you've for install_client_interceptors? ie reduce the level of indentation

Comment (debosmit) [reply opentracing_instrumentation/client_hooks/__init__.py]: of course -- doing that :)

File: opentracing_instrumentation/client_hooks/__init__.py [hunk 4]
```diff
@@ -60,14 +64,39 @@ def install_patches(patchers='all'):
...
-    raise ValueError('patchers argument must be None, "all", or a list')
+    if not _valid_args(patchers):
+        raise ValueError('patchers argument must be None, "all", or a list')
+
+    for patch_func_name in patchers:
+        logging.info('Loading client hook %s', patch_func_name)
+        patch_func = _load_symbol(patch_func_name)
+        logging.info('Applying client hook %s', patch_func_name)
+        patch_func()
+    return
```
Comment (black-adder) [inline opentracing_instrumentation/client_hooks/__init__.py]: rm return

**Pull request merged**